### ISSUE TITLE
Promoting an unchanged pre-release doesn't work

### ISSUE THREAD

**Issue by SamVerschueren** [lerna/lerna#2143]
## Expected Behavior
I'm using pre-release versions to test things out before releasing the real deal. Between a pre-release and the regular release, I don't have any extra changes. So when I run `lerna version`, I'd expect to be able to re-release my prereleased packages but then with the release tag instead. 

## Current Behavior
The docs of [@lerna/version](https://github.com/lerna/lerna/tree/master/commands/version#prerelease) describe the following

> If you have any packages with a prerelease version number (e.g. `2.0.0-beta.3`) and you run lerna version with and a non-prerelease bump (`major`, `minor`, or `patch`), it will publish those previously pre-released packages as well as the packages that have changed since the last release.

So it looks like running `lerna version patch` or something, should do the trick. But instead, it returns with the following message

> Current HEAD is already released, skipping change detection.

## Possible Solution
It would be nice if lerna could filter out the prereleased packages when detecting the changes.

## Steps to Reproduce (for bugs)

1.
2.
3.
4.

<details><summary>lerna.json</summary><p>

```json
{
  "packages": [
    "packages/*",
    "services/*"
  ],
  "version": "independent",
  "command": {
    "version": {
      "allowBranch": ["master", "staging"]
    }
  }
}
```
</p></details>

## Context
In our setup we have a `development`, `staging` and `master` branch. Every PR gets merged into `development` which deploys our code in a development environment. When the feature is done it is merged to `staging` and versioned with as pre-release, the code is then deployed to a staging/test environment. When everything works as expected, we merge things to `master`, version the packages as release and it gets pushed to production.

We are looking at Lerna for our monorepo setup and see if we can manage our releases like I described above.

## Your Environment

| Executable | Version |
| ---: | :--- |
| `lerna --version` | 3.15.0 |
| `npm --version`  | 6.8.0 |
| `node --version` | 10.15.1 |

| OS | Version |
| --- | --- |
| macOS High Sierra | 10.13.6 |

**Comment by SamVerschueren** [comment]
Apparently this command does the trick

```
# force all prerelease packages to be graduated
lerna version --conventional-commits --conventional-graduate
```

Although I'm not using conventional commit messages 🤔? Is there any reason why this works? Using the command without `--conventional-commits` doesn't work but it feels kinda weird to enable a flag while I don't use conventional commit messages.

**Comment by evocateur** [comment]
Yeah, I forget why we didn't go with a more general `--graduate-prerelease` or something like that. I could have sworn that's what _already_ happens in the non-`--conventional-commits` case...

**Comment by jshearer** [comment]
This workflow with `lerna version --conventional-commits --conventional-graduate` seems to... create extra CHANGELOG files and immediately commit them, which is very much not what I want, so I have to go back and rewrite the commit to remove those. Is there any way around this?

**Comment by siawo** [comment]
`lerna version --conventional-commits --no-changelog --conventional-graduate` This will be a workaround if you don't need the extra CHANGELOG files.
